pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1096,11 +1116,18 @@ func (og *operationGenerator) GenerateUnmapDeviceFunc(
 			}
 		}
 
-		// Execute tear down device
-		unmapErr := blockVolumeUnmapper.TearDownDevice(globalMapPath, deviceToDetach.DevicePath)
+		// Execute unmap device
+		unmapErr := blockVolumeUnmapper.UnmapDevice(globalMapPath, deviceToDetach.DevicePath)
```

Comment:
should we call UnmapDevice in the per-pod UnmapVolume instead? Then what if we had all the in-tree plugins called og.blkUtil.UnmapDevice in their plugin.UnmapDevice, and CSI calls Unpublish in its plugin.UnmapDevice

Reply:
Thank you for pointing it out.
This will be a big change, so I wanted to make an agreement before go ahead.

I will try implementing this way and update the commit.